File: validPalindrome.py
class Solution:
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """

        # front = 0
        # last = len(s)-1
        # while front < last:
        #     tempFC = s[front]
        #     tempLC = s[last]
        #     if ("a" <= tempFC <= "z" or "A" <= tempFC <= "Z" or "0" <= tempFC <= "9") == False:
        #         front += 1
        #         continue
        #     if ("a" <= tempLC <= "z" or "A" <= tempLC <= "Z" or "0" <= tempLC <= "9") == False:
        #         last -= 1
        #         continue
        #     if tempFC.lower() == tempLC.lower():
        #         front += 1
        #         last -= 1
        #     else:
        #         return False
        # return True

        lows = s.lower()
        cacheArr = []
        for c in lows:
            if "a" <= c <= "z" or "0" <= c <= "9":
                cacheArr.append(c)
        front = 0
        last = len(cacheArr)-1
        while front < last:
            if cacheArr[front] == cacheArr[last]:
                front += 1
                last -= 1
            else:
                return False
        return True


print(Solution().isPalindrome("A man, a plan, a canal: Panama"))
# print(Solution().isPalindrome("OP"))

File: validPalindrome.swift
class Solution {
    func isPalindrome(_ s: String) -> Bool {
        let tempS = s.lowercased()
        var cacheArr = [Character]()
        tempS.forEach { (c) in
            if ("a" <= c && c <= "z") || ("0" <= c && c <= "9") {
                cacheArr.append(c)
            }
        }
        let count = cacheArr.count
        var front = 0
        var end = count-1
        while front < end {
            let fc = cacheArr[front]
            let lc = cacheArr[end]
            if fc == lc {
                front += 1
                end -= 1
            } else {
                return false
            }
        }
        return true
    }
}